src/snowflake/snowpark/_internal/analyzer/snowflake_plan_node.py

```diff
@@ -159,17 +159,23 @@ class SaveMode(Enum):
     TRUNCATE = "truncate"
 
 
+class TableCreationSource(Enum):
+    EXPLICIT_USER_COMMAND = "explicit_user_command"
+    CACHE_RESULT = "cache_result"
```

Comment:
add comment here about what does those source mean, for example, CACHE_RESULT indicates table created by the df.cache_result API ect

Reply:
added